### ISSUE TITLE
Suppressions for react-no-dangerous-html require absolute path

### ISSUE THREAD

**Issue by kevinwilde** [Microsoft/tslint-microsoft-contrib#537]
Suppressions for react-no-dangerous-html only work when the file attribute contains the absolute path to the file.

I believe the problem is line 120 in `src/reactNoDangerousHtmlRule.ts` when the file attribute is compared for equality to a `SourceFile.fileName` which is an absolute path.

**Comment by JoshuaKGoldberg** [comment]
Agreed, and thanks for reporting this!

https://github.com/Microsoft/tslint-microsoft-contrib/blob/7e87ab24758fef21611d13091e74c55c4fc70a4e/src/reactNoDangerousHtmlRule.ts#L120

Should be a relatively simpler bug fix to switch that logic to checking if the paths generally match up. `/foo/bar/baz.tsx` should match with `C:/Code/Project/foobar/baz.tsx`.

**Comment by kevinwilde** [comment]
I can work on this. I'm thinking it could be changed to 
```typescript
path.resolve(exception.file) === this.getSourceFile().fileName
```

https://nodejs.org/api/path.html#path_path_resolve_paths
